### PR TITLE
Add thx.Arrays.map static wrapper

### PR DESCRIPTION
**Pull request by andywhite37**

### CODE REVIEW

File: src/thx/Arrays.hx [hunk 1]
```diff
@@ -622,6 +622,21 @@ It returns the last element of the array or null if the array is empty.
     return array[array.length-1];
 
 /**
+Static wrapper for `Array` `map` function.
+**/
+  #if js inline #end
```
Comment (fponticelli) [inline src/thx/Arrays.hx]: I would not specialize for `JS`. Native `map` is more likely to be slower that this map.

Comment (andywhite37) [inline src/thx/Arrays.hx]: I just copied the code for mapi for this if that makes any difference

Comment (fponticelli) [inline src/thx/Arrays.hx]: We should probably do the same for mapi :)

Comment (sledorze) [inline src/thx/Arrays.hx]: Has it been measured?

Comment (fponticelli) [inline src/thx/Arrays.hx]: A long time ago ... things might be different now.

Comment (fponticelli) [inline src/thx/Arrays.hx]: Added a test here: https://github.com/fponticelli/thx.benchmark/blob/master/demo/MapVsNativeMap.hx

In nodejs I get:

```
┃ description ┃    ops/sec ┃  error ┃ samples ┃ performance ┃
┃━━━━━━━━━━━━━┃━━━━━━━━━━━━┃━━━━━━━━┃━━━━━━━━━┃━━━━━━━━━━━━━┃
┃ native map  ┃  3,252,165 ┃ ±1.48% ┃      46 ┃  80% slower ┃
┃ static map  ┃ 16,648,575 ┃ ±1.17% ┃      42 ┃     fastest ┃
```

Quite a difference.

Comment (andypellucid) [inline src/thx/Arrays.hx]: I just did a little test too:

```
|| ┃ description       ┃    ops/sec ┃  error ┃ samples ┃ performance ┃
|| ┃━━━━━━━━━━━━━━━━━━━┃━━━━━━━━━━━━┃━━━━━━━━┃━━━━━━━━━┃━━━━━━━━━━━━━┃
|| ┃ thx.Arrays.mapi   ┃  2,774,280 ┃ ±0.96% ┃      95 ┃  85% slower ┃
|| ┃ native map with i ┃  2,740,368 ┃ ±0.91% ┃      91 ┃  85% slower ┃
|| ┃ indexed for loop  ┃ 18,838,296 ┃ ±0.98% ┃      94 ┃     fastest ┃
|| ┃ iterator for loop ┃ 18,412,222 ┃ ±1.33% ┃      79 ┃   2% slower ┃
```

Comment (andypellucid) [inline src/thx/Arrays.hx]: I'll go ahead and remove the native map usage... seems to be significantly slower than the for loop.

**Pull request merged**